File: chapter_5/Program.java
	class orderFood 
	{
	   	public static void main(String[] args)
		{ 	// declare fields n objects
			String sMenuItem;
			double dTotal;
			int iItems = 1;
			gatherBill oBill = new gatherBill ();
			displayMenu oMenu = new displayMenu ();
			while (true) 
			{// menu display loop with input loop & questions
				sMenuItem = oMenu.getChoice();
				if ( sMenuItem.equals("x") ) break;
				System.out.println("How many would you like to order?");
				iItems = Integer.parseInt(System.console().readLine());
				oBill.setCost ( sMenuItem, iItems ); 
			}// compute requested information
			dTotal = oBill.getCost(0.06);
			salestax oTax = new salestax();
			dTotal = oTax.round(dTotal, 2);// previous rounding function from earlier chapter
			System.out.print("Your total bill, including 6% sales tax, is: $");
			System.out.println(dTotal);// provide output 
		}
	}
	class displayMenu      
	{
		String sChoice = "x";
		public String getChoice ( ) 
		{// print menu
		
			System.out.println ("Yum Yum Burger Joint Lunch / Dinner menu:");
			System.out.println ("[A] Yum Yum Burger: $0.99 dollars");
			System.out.println ("[B] Grease Yum Fries: $0.79 dollars");
			System.out.println ("[C] Soda Yum: $1.09 dollars");
			System.out.println ("[X] Press x when done.");
			System.out.println ("Enter in one of the following: [a,b,c,x]:");
			sChoice = System.console().readLine ();
			while ( true ) 
			{ // check choices
				if ( sChoice.equals("a") || sChoice.equals("b") || sChoice.equals("c") || sChoice.equals("x") ) return sChoice;
				System.out.println ("Please enter in [a,b,c,x]:");
				sChoice = System.console().readLine ();
			}
		}
	}
	class gatherBill
	{
		double dCost;
		public gatherBill ( )
		{// constructor to preserve total cost and initialize
			if ( dCost > 0  )
			{
				// do nothing
			}
			else
			{
				dCost = 0;
			}
			
		}
		public double setCost ( String sItem, int iOrdered ) 
		{// prices
			if ( sItem.equals("a") ) dCost = dCost + 0.99 * iOrdered;
			if ( sItem.equals("b") ) dCost = dCost + 0.79 * iOrdered;
			if ( sItem.equals("c") ) dCost = dCost + 1.09 * iOrdered;
			System.out.println ("You added an item to your bill.");
			return dCost;
		}
		public double getCost ( double dTax )
		{// compute tax
			dCost = dCost + dCost * dTax;
			System.out.println ("You where taxed!");
			return dCost;
		}
	} 

File: chapter_5/Program.cs
using System;// change first class to main method & change setter return value
namespace hello_world
{
	class MainClass
	{
		public static void Main (string[] args)
		{
			string sMenuItem;
			double dTotal;
			int iItems = 1;
			gatherBill oBill = new gatherBill ();
			displayMenu oMenu = new displayMenu ();
			while (true) 
			{
				sMenuItem = oMenu.getChoice();
				if ( sMenuItem == "x" ) break;
				Console.WriteLine("How many would you like to order?");
				iItems = Int32.Parse(Console.ReadLine());
				oBill.setCost ( sMenuItem, iItems ); 
			}
			dTotal = oBill.getCost(0.06);
			dTotal = Math.Round(dTotal, 2);
			Console.Write("Your total bill, including 6% sales tax, is: $");
			Console.WriteLine(dTotal);
		}
	}
	class displayMenu      
	{
		string sChoice = "x";
		public string getChoice ( ) 
		{
		
			Console.WriteLine ("Yum Yum Burger Joint Lunch / Dinner menu:");
			Console.WriteLine ("[A] Yum Yum Burger: $0.99 dollars");
			Console.WriteLine ("[B] Grease Yum Fries: $0.79 dollars");
			Console.WriteLine ("[C] Soda Yum: $1.09 dollars");
			Console.WriteLine ("[X] Press x when done.");
			Console.WriteLine ("Enter in one of the following: [a,b,c,x]:");
			sChoice = Console.ReadLine ();
			while ( true ) 
			{
				if ( sChoice == "a" || sChoice == "b" || sChoice == "c" || sChoice == "x" ) return sChoice;
				Console.WriteLine ("Please enter in [a,b,c,x]:");
				sChoice = Console.ReadLine ();
			}
		}
	}
	class gatherBill
	{
		double dCost;
		public gatherBill ( )
		{
			if ( dCost > 0  )
			{
				// do nothing
			}
			else
			{
				dCost = 0;
			}
			
		}
		public double setCost ( string sItem, int iOrdered ) 
		{
			if ( sItem == "a" ) dCost = dCost + 0.99 * iOrdered;
			if ( sItem == "b" ) dCost = dCost + 0.79 * iOrdered;
			if ( sItem == "c" ) dCost = dCost + 1.09 * iOrdered;
			Console.WriteLine ("You added an item to your bill.");
			return dCost;
		}
		public double getCost ( double dTax )
		{
			dCost = dCost + dCost * dTax;
			Console.WriteLine ("You where taxed!");
			return dCost;
		}
	} 
}

